explainability-core/src/main/java/org/kie/trustyai/metrics/drift/ks_test/ApproxKSTest.java

```
@@ -0,0 +1,146 @@
+                // build epsilon sketch for given column
+                GKSketch sketch = new GKSketch(eps); 
+                dfTrain.getColumn(i).stream().mapToDouble(Value::asNumber).forEach(sketch::insert);
+                sketches.put(dfTrain.getColumnNames().get(i), sketch);
+             }
+        }
+        return new ApproxKSFitting(sketches);
+    }
+    /*
+     * Returns HypothesisTestResult per column with Approximate KSTest statistic, p-value and reject
+     */
+    public HashMap<String, HypothesisTestResult> calculate(Dataframe dfTest, double signif)
+    {
+        if(dfTest.getRowDimension() < 2)  {
+            throw new IllegalArgumentException( String.format("Passed dataframe is too small to calculate statistical hypotesis test."));
+        }
+        List<Type> types = dfTest.getColumnTypes();
+        List<String> testNames = dfTest.getColumnNames();
+
+        HashMap<String, HypothesisTestResult> result = new HashMap<>();
+        for (int i = 0; i < dfTest.getColumnDimension(); i++) {
+            if (types.get(i).equals(Type.NUMBER)) {
+                String colName = testNames.get(i);
+                if (!trainGKSketches.containsKey(colName)) {
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Passed dataframe not compatible with the train dataframe: no such column in train dataframe with name %s.",
+                                    testNames.get(i)));
+                }
+                GKSketch testSketch = new GKSketch(eps);
+                GKSketch trainSketch = trainGKSketches.get(colName);
+                dfTest.getColumn(i).stream().mapToDouble(Value::asNumber).forEach(testSketch::insert);
+                double d = 0.0;
+                try {
+                    d = computeKSDistance(trainSketch, testSketch);
+                } catch (GKException e) {
+                    e.printStackTrace();
```

Can we throw an error here as well? Maybe `RuntimeException`?